git-p4.py

```diff
@@ -769,13 +769,19 @@ def p4CmdList(cmd, stdin=None, stdin_mode='w+b', cb=None, skip_info=False,
                 #   - `depotFile[0-9]*`, `path`, or `clientFile` which may contain non-UTF8 encoded text
                 decoded_entry = {}
                 for key, value in entry.items():
-                    key = key.decode()
+                    key = key.decode('utf-8', 'backslashreplace')
                     if isinstance(value, bytes) and not (key in ('data', 'path', 'clientFile') or key.startswith('depotFile')):
-                        value = value.decode()
+                        try:
+                            value = value.decode('utf-8', 'backslashreplace')
+                        except UnicodeDecodeError as e:
+                            if key == 'desc':
+                                value = str(value)[2:-1]  # work around unicode decode errors in description (hex escapes will just be printed as \x7a for instance.
+                            else:
+                                raise e
```

Comment:
This hunk, as well as the ones immediately preceding and following it, probably want to live in their own commit. Maybe even in their own PR.